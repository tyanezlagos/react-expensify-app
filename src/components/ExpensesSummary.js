import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const total = numeral(expensesTotal / 100).format('$0,00.00');
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'; 
    return  (
        <div>
            <p>Viewing {expenseCount} {expenseWord} totalling {total}</p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getTotalExpenses(visibleExpenses) 
    };
} 

export default connect(mapStateToProps)(ExpensesSummary);
