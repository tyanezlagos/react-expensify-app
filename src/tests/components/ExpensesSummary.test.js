import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary correctly with single expense', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={11012} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={154222} />);
    expect(wrapper).toMatchSnapshot();
});