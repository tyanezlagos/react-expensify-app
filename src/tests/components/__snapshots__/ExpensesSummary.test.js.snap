// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     
    expenses
     totalling 
    $1,542.22
  </p>
</div>
`;

exports[`should render ExpensesSummary correctly with single expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $110.12
  </p>
</div>
`;
